crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -609,6 +609,7 @@ impl OlmMachine {
     /// this method between sync requests.
     ///
     /// [`mark_request_as_sent`]: #method.mark_request_as_sent
+    #[instrument(skip(users))]
```

Comment:
You used `skip(self)` elsewhere but not in the methods in this file. Do you really want to debug-print the `OlmMachine`? It's also possible to do `fields(foo = self.foo())` in the `instrument` arguments.

Reply:
I did `skip(self)` elsewhere because the `Debug` outputs of `IdentityManager` and `SessionManager` are firehoses. `OlmMachine` itself actually has a sensible `Debug` impl (it just prints the `user_id` and `device_id`) so it wasn't an issue.

> Do you really want to debug-print the `OlmMachine`? 

I don't know, tbh. It occurred to me it might be useful in a future application where there are multiple matrix clients (eg, for multi-account support). But in short, I have no strong feelings on the matter. Let me know what you prefer!

Reply:
I have no strong feelings about this either, but kinda expected the `Debug` impl to also be "a firehose", as you put it. wdyt @poljar @bnjbvr ?

Reply:
I think the problem here will be that EX already may log the user/device IDs and logging the whole `OlmMachine` will duplicate things again.

So unless we're sure we need it, I would leave it out.